From d3a82dba792d280dd39c9d2f3d4b8e01faf7ab7c Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Sun, 20 Mar 2022 12:50:50 +0200
Subject: [PATCH] [SW-57547] gaudi3: rom: use validation pass flag

In order to avoid redundant processing validation pass flag
is used to make sure that the information is refreshed only
when image is validated. Once the image is validated we do not
need to re-verify or update the relavent data on every cycle.

Change-Id: Ic6bb9540b12211163ac83e5dfdaa51f985521de2
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 13f6d79..1ce6d7d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -461,7 +461,7 @@
 	uint32_t img_plus_key_len;
 	uint8_t crc_bytes, num_chunks;
 
-	if (!security_efuse_image_validation_en())
+	if ((arcboot_rom_global.validation_pass) || (!security_efuse_image_validation_en()))
 		return;
 
 	boot_ptr_addr = boot_desc_param_get(&boot_ptr);
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 5fb4213..7947c80 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -316,9 +316,18 @@
 	security_load_fw_version();
 	security_load_keys();
 
-	boot_info_set(HL_BOOT_INFO_ROM_SECURITY_DATA, &arcboot_rom_global.sec_data,
-			sizeof(arcboot_rom_global.sec_data));
-	boot_info_finalize();
+	/*
+	 * efuse data is fixed. The only scenario of security data
+	 * being updated is during fake fuse update which would be
+	 * done through spi update, hence can be coupled with the
+	 * validation pass flag as every spi update would require
+	 * re-validation of the spi images
+	 */
+	if (!arcboot_rom_global.validation_pass) {
+		boot_info_set(HL_BOOT_INFO_ROM_SECURITY_DATA, &arcboot_rom_global.sec_data,
+				sizeof(arcboot_rom_global.sec_data));
+		boot_info_finalize();
+	}
 }
 
 int security_efuse_image_validation_en(void)
